pallets/author-reward-dest/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+		/// The inherent set author reward dest has already been submitted.
+		AlreadySet,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Set the author reward destination for the block.
+		#[pallet::call_index(0)]
+		#[pallet::weight((
+			T::DbWeight::get().writes(1),
+			DispatchClass::Mandatory
+		))]
+		pub fn set_author_reward_dest(origin: OriginFor<T>, dest: T::AccountId) -> DispatchResult {
+			ensure_none(origin)?;
+			ensure!(!<Author<T>>::exists(), Error::<T>::AlreadySet);
+			<Author<T>>::put(dest);
+			Ok(())
+		}
+	}
+
+	#[pallet::inherent]
+	impl<T: Config> ProvideInherent for Pallet<T> {
+		type Call = Call<T>;
+		type Error = InherentError;
+		const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
+
+		fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+			let inherent_data = data
+				.get_data::<InherentType>(&INHERENT_IDENTIFIER)
+				.expect("author reward dest inherent data not correctly encoded")
+				.expect("author reward dest inherent data must be provided");
+			Some(Call::set_author_reward_dest {
+				dest: T::AccountIdType::from(inherent_data).into(),
+			})
+		}
+
+		fn check_inherent(_call: &Self::Call, _data: &InherentData) -> Result<(), Self::Error> {
+			Ok(())
+		}
+
+		fn is_inherent(call: &Self::Call) -> bool {
+			matches!(call, Call::set_author_reward_dest { .. })
+		}
+	}
+
+	#[pallet::storage]
+	/// Author of current block.
+	pub(super) type Author<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
+}
+
+impl<T: Config> Pallet<T> {
+	/// Fetch the author of the block.
+	///
+	/// This is safe to invoke after inherent extrinsics and before `on_finalize`
+	pub fn author() -> Option<T::AccountId> {
+		<Author<T>>::get()
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate as pallet_author_reward_dest;
+	use frame_support::{
+		derive_impl,
+		inherent::{InherentData, ProvideInherent},
+		traits::{OnFinalize, UnfilteredDispatchable},
+	};
+	use sp_runtime::{traits::IdentityLookup, AccountId32, BuildStorage};
+
+	type Block = frame_system::mocking::MockBlock<Test>;
+
+	frame_support::construct_runtime!(
+		pub enum Test
+		{
+			System: frame_system,
+			AuthorRewardDest: pallet_author_reward_dest,
+		}
+	);
+
+	#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
+	impl frame_system::Config for Test {
+		type AccountId = AccountId32;
+		type Lookup = IdentityLookup<Self::AccountId>;
+		type Block = Block;
+	}
+
+	impl pallet::Config for Test {
+		type AccountIdType = AccountId32;
+	}
+
+	fn new_test_ext() -> sp_io::TestExternalities {
+		let t = frame_system::GenesisConfig::<Test>::default().build_storage().unwrap();
+		t.into()
+	}
+
+	#[test]
+	fn sets_author_lazily() {
+		new_test_ext().execute_with(|| {
+			System::reset_events();
+
+			let mut inherent_data = InherentData::new();
+			AuthorRewardDest::is_inherent_required(&inherent_data)
+				.expect("Must be OK")
+				.expect("Is mandatory");
```

Comment:
tests are failing here

@ggwpez @thiolliere seems like it should just be removed? since you don't want to use `is_inherent_required`

Reply:
Yes, I added a different test instead.